torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -269,14 +271,34 @@ void ProcessGroupAgent::sync() {
 }
 
 void ProcessGroupAgent::start() {
+  rpcRunning_.store(true);
   listenerThread_ = std::thread(&ProcessGroupAgent::listenLoop, this);
   futureTimeoutThread_ =
       std::thread(&ProcessGroupAgent::pollTimedOutRPCs, this);
 }
 
```

Comment:
What happened to the destructor? I don't see it in the latest version.

Reply:
The destructor tests aren't passing, and I'm debugging why. But for now I've removed it and the associated test in this PR, since it's not a strict requirement to be able to shutdown PG agent.